shell/browser/api/electron_api_app.cc

```diff
@@ -1071,31 +1070,34 @@ Browser::LoginItemSettings App::GetLoginItemSettings(
 }
 
 #if defined(USE_NSS_CERTS)
-void App::ImportCertificate(const base::DictionaryValue& options,
+void App::ImportCertificate(base::Value options,
                             net::CompletionRepeatingCallback callback) {
+  DCHECK(options.is_dict());
+
   auto browser_context = ElectronBrowserContext::From("", false);
   if (!certificate_manager_model_) {
-    auto copy = base::DictionaryValue::From(
-        base::Value::ToUniquePtrValue(options.Clone()));
     CertificateManagerModel::Create(
         browser_context.get(),
         base::BindRepeating(&App::OnCertificateManagerModelCreated,
```

Comment:
The `CreationCallback` used by `CertificateManagerModel::Create` is currently a repeating callback for no reason, the callback is passed to other threads and which always bind as `OnceCallback` , it would be better to upstream a CL that changes `CreationCallback` to `OnceCallback`.

```
typedef base::Callback<void(std::unique_ptr<CertificateManagerModel>)>
      CreationCallback;

// static
void CertificateManagerModel::Create(
    content::BrowserContext* browser_context,
    CertificateManagerModel::Observer* observer,
    const CreationCallback& callback);
```

to 

```
using CreationCallback = base::OnceCallback<void(std::unique_ptr<CertificateManagerModel>)>;

// static
void CertificateManagerModel::Create(
    content::BrowserContext* browser_context,
    CertificateManagerModel::Observer* observer,
    CreationCallback callback);
```

Reply:
@deepak1556 would you prefer i add a patch and implement this change in this PR or a follow-up?

Reply:
We maintain a copy of the `CertificateManagerModel` class https://github.com/electron/electron/blob/master/chromium_src/chrome/browser/certificate_manager_model.cc, you can make the changes in this PR.

I would also recommend uploading a similar patch to upstream `CertificateManagerModel` code.